proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+
+```xml
+	<struct name="AppServiceManifest">
+		<param name="serviceName" type="String" mandatory="false"/>
+		<param name="serviceId" type="String" mandatory="true"/>
+		<param name="serviceType" type="AppServiceType" mandatory="true">
+			<description>The type of service that is to be offered by this app</description>
+		</param>
+		
+		<param name="serviceIcon" type="String" mandatory="false">
+			<description> The file name of the icon to be associated with this service. Most likely the same as the appIcon.</description>
+		</param>
+		
+		<param name="allowAppConsumers" type="Boolean" mandatory="false"/>
+
+				
+		<param name="uriPrefix" type="String" mandatory="false"/>
+		<param name="uriActions" type="String" array="true" mandatory="false">
+		
+		<param name="serviceActive" type="Boolean" mandatory="false"/>
+		
+		<param name="rpcSpecVersion" type="SyncMsgVersion" mandatory="false"/>
+		<param name="handledRPCs" type="FunctionID" array="true" mandatory="false">
+			<description> This field contains the Function IDs for the RPCs that this service intends to handle correctly. This means the service will provide meaningful responses. </description>
+		</param>
+		
+		<param name="mediaServiceManifest" type="MediaServiceManifest" mandatory="false"/>
+		<param name="weatherServiceManifest" type="WeatherServiceManifest" mandatory="false"/>
+		<param name="navigationServiceManifest" type="NavigationServiceManifest" mandatory="false"/>
+		<param name="voiceAssistantServiceManifest" type="VoiceAssistantServiceManifest" mandatory="false"/>
+
+	</struct>
+	
+```
+
+Based on the specific type of service being offered, an additional manifest struct should be included. These structs are very specific to the service that is being offered. These are defined in the previous section  [Service Types](#service-types).
+
+
+##### Publishing the service - `PublishAppService`
+The next action the App Service has to take is publishing their service. This is performed after the `RegisterAppInterface` response is received. The `PublishAppService` request contains the previously created `AppServiceManifest` object that will inform the system of the capabilities and info about this service.
+
+
+```xml
+<function name="PublishAppService" functionID="PublishAppServiceID" messagetype="request">
+	<description>Registers a service offered by this app on the module</description>
+
+	<param name="appServiceManifest" type="AppServiceManifest" mandatory="false"/>
+
+	
+</function>
+
+
+<function name="PublishAppService" functionID="PublishAppServiceID" messagetype="response">
+	<description>Response to the request to register a service offered by this app on the module</description>
+
+	<param name="success" type="Boolean" platform="documentation" mandatory="true">
+		<description> true if successful; false, if failed </description>
```

Comment:
Should there be a comma after "true" or should the comma after "false" be removed so that these match?